src/libraries/System.Text.Json/gen/Model/ImmutableEquatableArray.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace System.Text.Json.SourceGeneration
+{
+    /// <summary>
+    /// Provides an immutable list implementation which implements sequence equality.
+    /// </summary>
+    public sealed class ImmutableEquatableArray<T> : IEquatable<ImmutableEquatableArray<T>>, IReadOnlyList<T>
```

Comment:
you might want to add a constraint like where T : IEquatable<T>, because the whole caching idea only works if you have proper structural equality and not the normal ref equality, as you wrote in the comment in the other file. I saw that you converted all those Specs to records, which auto-implement IEquality anyway.

Reply:
Good idea, thanks.